context/app/static/js/pages/Vis/Vis.jsx

```diff
@@ -33,55 +33,94 @@ import useSearchData from 'js/hooks/useSearchData';
 
  */
 
-const donorRaceQuery = {
+const donorRaceSexQuery = {
   size: 0,
-  query: {
-    bool: {
-      filter: {
-        term: {
-          entity_type: 'donor',
-        },
-      },
-    },
-  },
   aggs: {
-    'mapped_metadata.race': {
-      terms: {
-        field: 'mapped_metadata.race.keyword',
+    composite_data: {
+      composite: {
+        sources: [
+          {
+            'mapped_metadata.race': {
+              terms: {
+                field: 'mapped_metadata.race.keyword',
+              },
+            },
+          },
+          {
+            'mapped_metadata.blood_type': {
+              terms: {
+                field: 'mapped_metadata.blood_type.keyword',
+              },
+            },
+          },
+        ],
+        size: 10000,
       },
     },
   },
 };
 
 function Vis() {
   const { elasticsearchEndpoint, nexusToken } = useContext(AppContext);
-  const { searchData } = useSearchData(donorRaceQuery, elasticsearchEndpoint, nexusToken);
+
+  const { searchData } = useSearchData(donorRaceSexQuery, elasticsearchEndpoint, nexusToken);
   if (!('aggregations' in searchData)) {
     return null;
   }
-  const { buckets } = searchData.aggregations['mapped_metadata.race'];
+
+  function getCount(buckets, bloodType, race) {
+    const filtered = buckets.filter(
+      (b) => b.key['mapped_metadata.blood_type'] === bloodType && b.key['mapped_metadata.race'] === race,
+    );
+    return filtered.length ? filtered[0].doc_count : 0;
+  }
+  function getKeyValues(buckets, key) {
+    return [...new Set(buckets.map((b) => b.key[key]))];
```

Comment:
```suggestion
    return new Set(buckets.map((b) => b.key[key]));
```
This came up in the call: I think you can just return a set here, instead of converting it back to a list... but I could be wrong.

Reply:
You are right, we talked about it, I forgot to take it out.